test_cases/solidity/reentracy/modifier_reentrancy/modifier_reentrancy.sol

```diff
@@ -22,7 +22,9 @@ contract ModifierEntrancy {
 }
 
 contract Bank{
-    function supportsToken() external pure returns(bytes32){
+    uint state_var;
+    function supportsToken() external returns(bytes32){
+        state_var = 1;
```

Comment:
IMO this state_var is a bit of a distraction that reduces the clarity of the example. 
For the sake of the bug, all that's necessary is to remove the `pure` keyword. The compiler just emits a warning that visibility can be restricted. 

Do you think it makes sense to remove it?